### PR TITLE
Checked if teamData is empty before trying to convert it to Object

### LINKED ISSUE #273: [Bug] - Support page error without any teams in database
### Pre-requisites:
- [x] I looked through both [open and closed issues](../issues?utf8=✓&q=is%3Aissue) and did not find an existing bug.

## Description
When there are no teams there will always be an error on the support page

## Expected Behavior
No error, it should just say there are no teams

## Actual Behavior
Got an error

## Steps to Reproduce
Reset database, start app, visit support page

## Other Information
image

### PR DESCRIPTION
**Pull request by RazvanPreotu99**
## Pre-Requisites
- [x] Yes, I updated [Authors.md](../blob/main/Authors.md) **OR** this is not my first contribution
- [x] Yes, I included and/or modified tests to cover relevent code **OR** my change is non-technical
- [x] Yes, I wrote this code entirely myself **OR** I properly attributed these changes in [Third Party Notices](../blob/main/THIRD-PARTY-NOTICES.txt)

## Description of Changes

Checked if teamData is empty before converting to Object.

## Related Issues

Closes #273 - Support page error when there are no teams.
Co-authored by: Michael Nguyen <50305984+mkn170030@users.noreply.github.com>

### CODE REVIEW

File: src/components/RegisteredTeamsList.tsx [hunk 1]
```diff
@@ -42,65 +42,66 @@ export const RegisteredTeamsList: React.FC<RegisteredTeamsListProps> = ({ secret
...
-    teamData.forEach((team: Team) => {
-      Object.keys(team).forEach((key) => {
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        const val = (team as any)[key as keyof Team];
-        if (Array.isArray(val)) {
-          // eslint-disable-next-line no-param-reassign, @typescript-eslint/no-explicit-any
-          ((team as unknown) as any)[key] = val.join(', ');
-        }
+    if (teamData.length != 0) {
+      const keys = Object.keys(teamData[0]).filter((key) => !['id', 'name', 'members', 'tableNumber', 'projectDescription'].includes(key));
```
Comment (SpencerKaiser) [inline src/components/RegisteredTeamsList.tsx:46]:
```suggestion
      const keys = Object.keys(teamData[0] ?? {}).filter((key) => !['id', 'name', 'members', 'tableNumber', 'projectDescription'].includes(key));
```

File: src/components/RegisteredTeamsList.tsx [hunk 2]
```diff
@@ -42,65 +42,66 @@ export const RegisteredTeamsList: React.FC<RegisteredTeamsListProps> = ({ secret
...
-    const keys = Object.keys(teamData[0]).filter((key) => !['id', 'name', 'members', 'tableNumber', 'projectDescription'].includes(key));
-    teamData.forEach((team: Team) => {
-      Object.keys(team).forEach((key) => {
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        const val = (team as any)[key as keyof Team];
-        if (Array.isArray(val)) {
-          // eslint-disable-next-line no-param-reassign, @typescript-eslint/no-explicit-any
-          ((team as unknown) as any)[key] = val.join(', ');
-        }
+    if (teamData.length != 0) {
```
Comment (SpencerKaiser) [inline src/components/RegisteredTeamsList.tsx:45]: Remove this and then save to get the indentation back to normal

**Review by SpencerKaiser — CHANGES_REQUESTED** [review]
Good catch! There's a slightly more elegant way of handling this, which is using the `??` syntax to swap what would have been `undefined` for an empty object instead. The end result is the same, but we avoid having to indent a big chunk of the code and having to add a new `if`.

The thing I described is called the [`nullish coalescing operator`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#nullish-coalescing) but here's the gist:

`teamData[0] ?? {}` basically equates to `teamData[0] === undefined ? teamData[0] : {}`

File: src/components/RegisteredTeamsList.tsx [hunk 3]
```diff
@@ -42,7 +42,8 @@ export const RegisteredTeamsList: React.FC<RegisteredTeamsListProps> = ({ secret
     }
 
     const teamData: Team[] = await res.json();
-    const keys = Object.keys(teamData[0]).filter((key) => !['id', 'name', 'members', 'tableNumber', 'projectDescription'].includes(key));
+
```
Comment (johnkahn) [inline src/components/RegisteredTeamsList.tsx:45]:
```suggestion
```

File: src/components/RegisteredTeamsList.tsx [hunk 4]
```diff
@@ -100,7 +101,7 @@ export const RegisteredTeamsList: React.FC<RegisteredTeamsListProps> = ({ secret
         sortable: true,
       },
     ];
-
+
```
Comment (johnkahn) [inline src/components/RegisteredTeamsList.tsx:104]:
```suggestion

```

File: src/components/RegisteredTeamsList.tsx [hunk 5]
```diff
@@ -127,8 +128,8 @@ export const RegisteredTeamsList: React.FC<RegisteredTeamsListProps> = ({ secret
               {teams.length > 0 ? (
                 <DataTable columns={columns} data={teams} expandableRows expandableRowsComponent={<ExpandableRow />} />
               ) : (
-                'No teams registered yet... check back soon!'
-              )}
+                  'No teams registered yet... check back soon!'
+                )}
```
Comment (johnkahn) [inline src/components/RegisteredTeamsList.tsx:132]:
```suggestion
                'No teams registered yet... check back soon!'
              )}
```

**Review by johnkahn — CHANGES_REQUESTED** [review]
Looks like some weird spacing got committed, you should be able to click `Add suggestion to batch` on each of these and commit them all at once 👍

**Review by johnkahn — DISMISSED** [review]
Thanks for doing this, we didn't realize it would be such a short fix 🙈 We'll make sure we have some larger ones going forward.

**Review by johnkahn — APPROVED** [review]

**Review by SpencerKaiser — APPROVED** [review]
Nice catch!

**Pull request merged**